File: GeeksForGeeks/Basic/Distance_and_displacement.cpp
#include <iostream>
#include <cmath>
using namespace std;

int findDistance(int arr[], int n, string direction) {
    int distance = 0;
    int point[2] = {0, 0};
    for(int i = 0; i < n; ++i) {
        distance += arr[i];
        switch (direction[i]) {
          case 'N': point[1] += arr[i];
                    break;
          case 'S': point[1] -= arr[i];
                    break;
          case 'E': point[0] += arr[i];
                    break;
          case 'W': point[0] -= arr[i];
                    break;
        }
    }
    distance += ceil(sqrt(point[0] * point[0]
                      + point[1] * point[1]));
    return distance;
}

int main() {
    ios::sync_with_stdio(false);
    int test_case;
    int m;
    string str;
    cin >> test_case;
    while (test_case--) {
        cin >> m;
        int arr[m];
        for(int i = 0; i < m; ++i) {
            cin >> arr[i];
        }
        cin >> str;
        cout << findDistance(arr, m, str) << endl;
    }
}

File: GeeksForGeeks/Basic/Distance_and_displacement.py
#Python 2.7

import math

def find_distance(a, n, s):
    distance = 0
    point = [0, 0]
    for i in range(len(a)):
        distance += a[i]
        if s[i] == 'N':
            point[1] += a[i]
        elif s[i] == 'S':
            point[1] -= a[i]
        elif s[i] == 'E':
            point[0] += a[i]
        else:
            point[0] -= a[i]
    return distance + int (math.ceil(math.sqrt(point[0]**2 + point[1]**2)))

if __name__ == '__main__':
    test_case = input()
    while test_case:
        m = input()
        arr = [int(x) for x in raw_input().split()]
        direction = raw_input()
        print find_distance(arr, m, direction)
        test_case -= 1
